nucypher/config/base.py

```diff
@@ -402,10 +404,12 @@ def __init__(self,
         self.is_me = True
         self.checksum_address = checksum_address
 
-        # Keyring
+        # Keystore
         self.crypto_power = crypto_power
-        self.keyring = keyring or NO_KEYRING_ATTACHED
-        self.keyring_root = keyring_root or UNINITIALIZED_CONFIGURATION
+        if keystore_path and not keystore:
+            keystore = Keystore(keystore_path=keystore_path)
+        self.__keystore = self.__keystore = keystore or NO_KEYSTORE_ATTACHED.bool_value(False)
```

Comment:
`self.__keystore = self.__keystore` - a typo? How does that not raise `AttributeError`?

Reply:
Yes, definitely a typo and good question!

Reply:
Ah, I suppose the second assignment gets executed first.